enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,224 @@
+type EgressIPStatusItem struct {
+// Assigned node name
+Node string `json:"node"`
+// Assigned egress IP
+EgressIP string `json:"egressIP"`
+// Route via host network or OVN
+RouteViaHost bool `json:"routeViaHost"`
+}
+```
+
+This will aid debugging of egress IP assignments but also aid egress IP controller located within ovnkube node to not
+have to recalculate if the egress IP is primary or secondary networks.
+
+### Risks and Mitigations
+- OVN kube node must recalculate all the matched pods for a given egress IP. This will require refactoring this logic
+  from cluster manager and may introduce bugs to egress IP itself.
+- Stale EgressIPs assigned to a secondary network interface
+  A component will be created to manage all network interface address operations.
+  This component will independently ensure the state of all network interfaces is what we expect.
+  Egress IP(s) assigned to an interface will contain a well-known label which will allow us to clean up stale egress IP addresses.
+  Therefore, we don't need any internal caches for this.
+- Stale linux routing rules in the routing policy database control
+  A component will be created to manage all rule operations.
+  We can ensure stale routes are cleaned because we use a specific priority that only we manage, therefore we don't
+  need any internal caches for this.
+  This component will independently manage the operations of rules and ensure they match the state we want.
+- Stale iptables rules
+  A component will be created to manage all iptable operations.
+  This component will independently manage the operations of iptables and ensure they match the state we want.
+  This component will ensure the rules we expect within a newly created specific chain are what we expect.
+  Therefore, we don't need any internal caches for this because we know we manage all rules within this chain.
+- Incorrect egress IP network interface assignment
+  The host addresses label will determine which network (interface) the egress IP will be assigned to.
+  Users may not expect that we will assign this egress IP to whichever interface contains the longest prefix match [1].
+  This is not a problem because we will emit the packet to the correct network regardless.
+- Unable to determine if an egress IP belongs to a primary network or a secondary network
+  Primary network can be determined from well-known label `k8s.ovn.org/node-primary-ifaddr`
+  Therefore anything outside this subnet mask is consider a "secondary" network
+- What about secondary bridges managed by OVN aka br-ex1
+  We ignore them and do not support egress IP.
+- Selecting the wrong network types from interfaces
+  Only consider addresses with scope `Global`.
+
+[1] https://en.wikipedia.org/wiki/Longest_prefix_match
+
+## Design Details
+### Cluster Manager
+If egress IP is within a subnet of a secondary network, it will need to assign this egress IP to a node which hosts this network.
```

Comment:
>I actually expect multiple nodes to have interfaces on the same subnet; how will it choose a node from the available ones ?

We expose all the networks we find on a node and annotate the `host-addresses` annotation including the networks mask.  
Cluster manager then decides which node contains a network which can "host" the egress IP.
 We use longest prefix match to decide which is the optimum network to choose. If there are multiple nodes that match, then one of them will be chosen at random as the host to host the egress IP.

Reply:
OK, that's an option; thanks for explaining.

If possible, I would add this info to the doc. 

Would it make sense to have some sort of algorithm that ensures the number of egress IPs across nodes are on par instead of going "full" random ?

Reply:
Added to the doc.
yes, I need to revisit this algorithm with these changes and ensure we spread the load.